src/core/thread/mle.cpp

```diff
@@ -4304,14 +4304,22 @@ Error Mle::DetachGracefully(otDetachGracefullyCallback aCallback, void *aContext
     mDetachGracefullyCallback = aCallback;
     mDetachGracefullyContext  = aContext;
 
-    if (IsChild() || IsRouter())
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+    IgnoreError(Get<BorderRouter::RoutingManager>().SetEnabled(false));
+#endif
+
+    if (IsChild() || IsRouter() || IsLeader())
```

Comment:
If the device is router or child, Thread is stopped immediately upon receiving Address Release response or Child Update Response. Is it possible that routing manager doesn't have enough time to stop?